mantle/platform/api/gcloud/compute.go

```diff
@@ -35,32 +34,26 @@ func (a *API) vmname() string {
 	return fmt.Sprintf("%s-%x", a.options.BaseName, b)
 }
 
-// ["5G:interface=NVME"], by default the disk type is local-ssd
-func ParseDiskSpec(spec string, zone string) (*compute.AttachedDisk, error) {
-	split := strings.Split(spec, ":")
+// ["5G:channel=nvme"], by default the disk type is local-ssd
+func ParseDisk(spec string, zone string) (*compute.AttachedDisk, error) {
+	var diskmap map[string]string
+	var disksize int64
 	var diskinterface string
-	var disksize string
-	if len(split) == 1 {
-		disksize = split[0]
-	} else if len(split) == 2 {
-		disksize = split[0]
-		for _, opt := range strings.Split(split[1], ",") {
-			if strings.HasPrefix(opt, "interface=") {
-				diskinterface = strings.TrimPrefix(opt, "interface=")
-				if diskinterface == "" || (diskinterface != "NVME" && diskinterface != "SCSI") {
-					return nil, fmt.Errorf("invalid interface opt %s", opt)
-				}
-			} else {
-				return nil, fmt.Errorf("unknown disk option %s", opt)
+	var err error
+
+	if disksize, diskmap, err = util.ParseDiskSpec(spec); err != nil {
+		return nil, fmt.Errorf("failed to parse disk spec %s: %w", spec, err)
+	}
+	for key, value := range diskmap {
+		switch key {
+		case "channel":
+			if value != "nvme" && value != "scsi" {
```

Comment:
Very minor style thing, I think this would "read" better as:

```
switch value {
    case "nvme":
    case "scsi":
      break
    default:
      return nil, fmt.Errorf("Invalid channel: %q", value)
```

For the same reasons as we're doing a `switch key` - adding an element to the list is more obvious.